Mail2Bug/Email/EWS/EWSIncomingAttachmentConverter.cs

```diff
@@ -0,0 +1,134 @@
+
+namespace Mail2Bug.Email.EWS
+{
+    public class EWSIncomingAttachmentConverter
+    {
+        // TFS drops large base64 inline images: http://johannblais.blogspot.com/2015/08/migrate-tfs-repro-steps-field-to.html
+        private const int MaxBase64Length = 22 * 1024; // This a safe guess
+
+        private readonly EmailMessage _message;
+
+        private string _bodyText;
+        private List<IIncomingEmailAttachment> _attachments = null;
+
+        public EWSIncomingAttachmentConverter(EmailMessage message)
+        {
+            _message = message;
+        }
+
+        public void ProcessAttachments(bool convertInlineAttachments = false)
+        {
+            _bodyText = _message.Body.Text ?? String.Empty;
+            _attachments = new List<IIncomingEmailAttachment>();
+
+            bool doInlining = !String.IsNullOrEmpty(_bodyText) && convertInlineAttachments;
+
+            foreach (var ma in _message.Attachments)
+            {
+                if (doInlining)
+                    ma.Load(); // Content property requires load, and for RTF the ContentType property is also not available unless loaded
+
+                if (ma is FileAttachment)
+                    _attachments.Add(new EWSIncomingFileAttachment(ma as FileAttachment));
+                else if (ma is ItemAttachment)
+                    _attachments.Add(new EWSIncomingItemAttachment(ma as ItemAttachment));
+                else
+                    Logger.ErrorFormat("Skipping attachment because it's not a file attachment ({0})", ma.Name);
+            }
+
+            if (doInlining)
+            {
+                // RTF-email attachments have no ContentId value set. The only way to match them is by position.
+                // We want a uniform way to match attachments to content id references, so here we assign
+                // the applicable content id to each inline attachment by traversing the HTML for
+                // "cid:" references and positionally matching to the attachments.
+                // More info: https://blogs.msdn.microsoft.com/webdav_101/2016/06/21/ews-and-inline-attachments/
+                if (IsNativeRtfBody(_message))
+                    FixupContentIds();
+
+                // Use base64 when possible because these images will display inside Visual Studio from 
+                // hosted VSTS (visualstudio.com).
+                Base64EncodeImages();
+            }
+        }
+
+        public string BodyText { get { return _bodyText; } }
+
+        public IEnumerable<IIncomingEmailAttachment> Attachments { get { return _attachments; } }
+
+        private void FixupContentIds()
+        {
+            const string pattern = @"(\""cid:)(.*)(\"")";
+
+            var inlineAttachments = _attachments.Where(x => x.IsInline);
+
+            int index = 0;
+
+            _bodyText = Regex.Replace(_bodyText, pattern, match =>
+            {
+                var inlineAttachment = inlineAttachments.ElementAtOrDefault(index++);
```

Comment:
Minor style suggestion: Could we just do something like
inlineAttachment = inlineAttachments.ElementAtOrDefault(index++) as EWSIncomingFileAttachment and check that against null?

I'm assuming that null as AnyObjectType returns null, but can't verify right now (using a machine with no .NET environment installed)